static/js/fetch_status.js

```diff
@@ -30,7 +39,13 @@ const FetchStatus = function () {
         loading_newer = true;
     };
 
-    self.finish_newer_batch = function (opts) {
+    self.finish_newer_batch = function (messages, opts) {
+        const found_max_message_id = max_id_for_messages(messages);
+        if (opts.found_newest && expected_max_message_id > found_max_message_id) {
+            loading_newer = false;
+            found_newest = false;
+            return true;
```

Comment:
In the first commit version of this, it'd be better to just do an if/else statement rather than mutating `opts`.

Reply:
I'm also skeptical of the change to have this return something, especially given that it doesn't do so in the other code path.